metabulo/views.py

```diff
@@ -86,6 +89,55 @@ def upload_csv_file(file, meta):
         raise
 
 
+_excel_mime_types = [
+    'application/vnd.ms-excel',
+    'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet'
+]
+
+
+@csv_bp.route('/excel/upload', methods=['POST'])
+@use_kwargs({
+    'file': fields.Field(location='files', required=True,
+                         validate=lambda f: f.content_type in _excel_mime_types),
+    'meta': JSONDictStr(missing={})
+})
+def upload_excel_file(file: FileStorage, meta: Dict):
+    excel_sheets = pandas.read_excel(file, sheet_name=None)  # type: Dict[str, pandas.DataFrame]
+    excel_sheets = {sheet: data for (sheet, data) in excel_sheets.items() if not data.empty}
+
+    basename = PurePath(cast(str, file.filename)).with_suffix('')
```

Comment:
What's the purpose of the type casting?  Isn't `file.filename` already a string?

Reply:
it should be but according to mypy and the typings is isn't. Thus the type cast to satify the mypy linter.

Reply:
Sounds good.  At some point, I need to figure out how to get type validation working in my editor.